fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/lowcardinality/DecodeCollector.java

```diff
@@ -607,6 +613,48 @@ public DecodeInfo visitPhysicalOlapScan(OptExpression optExpression, DecodeInfo
         return info;
     }
 
+    private Pair<Boolean, Optional<ColumnDict>> checkConnectorGlobalDict(Table table, ColumnRefOperator column) {
+        // Condition 1:
+        if (!supportLowCardinality(column.getType())) {
+            return new Pair<>(false, Optional.empty());
+        }
+
+        if (!sessionVariable.isEnableArrayLowCardinalityOptimize() && column.getType().isArrayType()) {
+            return new Pair<>(false, Optional.empty());
+        }
+
+        // Condition 2: the varchar column is low cardinality string column
+        ColumnStatistic columnStatistic = GlobalStateMgr.getCurrentState().getStatisticStorage()
+                .getConnectorTableStatistics(table, List.of(column.getName())).get(0).getColumnStatistic();
+
+        if (!columnStatistic.isUnknown() &&
```

Comment:
we allow collecting dict for columns without stats?

Reply:
Yes, when there is statistics, we check ndv, if it's bigger than threshold, we don't collect dictionary.
when there is no statistics, we just collect dictionary, and it maybe need to be supplemented, but 
we have a version control, we only do several times in 7 day, and even over 7days, we not always 
supplement it again unless the dictionary is updated in background cache thread.